curriculum/locales/english/learn-digital-signatures-by-building-a-wallet.md

```diff
@@ -4,206 +4,5928 @@
+
+while (!hash.startsWith('0'.repeat(difficulty))) {
+  nonce++;
+  hash = sha256(nonce + previousBlock.hash + JSON.stringify(transactions)).toString();
+}
+
+const newBlock = {
+  hash,
+  nonce,
+  previousHash: previousBlock.hash,
+  transactions
+}
+
+const rewardAddress = getWalletAddressFromName('Me');
+const rewardTransaction = {
+  fromAddress: null,
+  toAddress: 'Me',
+  amount: 50
+}
+
+blockchain.push(newBlock);
+writeBlockchain(blockchain);
+writeTransactions([rewardTransaction]);
+```
+
+## 88
+
+### --description--
+
+Make sure you can see your blockchain and transactions files, and then re-initialize your blockchain so you don't have any transactions that use names.
+
+### --tests--
+
+You should run `node init-blockchain.js` in the terminal
+
+```js
+await new Promise(res => setTimeout(res, 1000));
+const lastCommand = await __helpers.getLastCommand();
+assert.equal(lastCommand.replace(/\s+/g, ' ').trim(), 'node init-blockchain.js');
+```
+
+Your `blockchain.json` file should be an array with one block (object)
+
+```js
+await new Promise(res => setTimeout(res, 1000));
+const fileContents = await __helpers.getJsonFile('learn-digital-signatures-by-building-a-wallet/blockchain.json');
+assert.isArray(fileContents);
+assert.lengthOf(fileContents, 1);
+```
+
+Your `transactions.json` file should be an empty array
+
+```js
+await new Promise(res => setTimeout(res, 1000));
+const fileContents = await __helpers.getJsonFile('learn-digital-signatures-by-building-a-wallet/transactions.json');
+assert.isArray(fileContents);
+assert.lengthOf(fileContents, 0);
+```
+
+### --seed--
+
+#### --"mine-block.js"--
+
+```js
+import sha256 from 'crypto-js/sha256.js';
+import { 
+  getBlockchain,
+  writeBlockchain,
+  getTransactions,
+  writeTransactions,
+  getWalletAddressFromName
+} from './blockchain-helpers.js';
+
+const blockchain = getBlockchain();
+const previousBlock = blockchain[blockchain.length - 1];
+const transactions = getTransactions();
+
+let nonce = 0;
+let hash = sha256(nonce + previousBlock.hash + JSON.stringify(transactions)).toString();
+const difficulty = 2;
+
+while (!hash.startsWith('0'.repeat(difficulty))) {
+  nonce++;
+  hash = sha256(nonce + previousBlock.hash + JSON.stringify(transactions)).toString();
+}
+
+const newBlock = {
+  hash,
+  nonce,
+  previousHash: previousBlock.hash,
+  transactions
+}
+
+const rewardAddress = getWalletAddressFromName('Me');
+const rewardTransaction = {
+  fromAddress: null,
+  toAddress: rewardAddress,
+  amount: 50
+}
+
+blockchain.push(newBlock);
+writeBlockchain(blockchain);
+writeTransactions([rewardTransaction]);
+```
+
+## 89
+
+### --description--
+
+Mine two new blocks to add some tokens to your blockchain.
+
+### --tests--
+
+You should run `node mine-block.js` in the terminal twice
+
+```js
+await new Promise(res => setTimeout(res, 1000));
+const lastCommand = await __helpers.getLastCommand();
+assert.equal(lastCommand.replace(/\s+/g, ' ').trim(), 'node mine-block.js');
+```
```

Comment:
It is not a big deal, because the lesson works as expected, but the test text does not match what the test is doing.